code/src/components/HappyThoughtsEntry.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react'
+
+import moment from 'moment'
+
+const HappyThoughtsEntry = ({ id, message, hearts, createdAt }) => {
+  const [likedByMe, setLikedByMe] = useState(false)
+  const [heartsCounter, setHearts] = useState(hearts)
+
+  const sendHeartToApi = () => {
+    fetch(`https://happy-thoughts-technigo.herokuapp.com/thoughts/${id}/like`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' }
+    }).then((res) => res.json())
+      .then((data) => {
+        // Fetch the number of hearts fresh from API
+        setHearts(data.hearts);
+      })
+  }
+
+  const addHeart = () => {
+    setLikedByMe(true)
+    // Skip posting you have already liked something once
+    if (!likedByMe) {
+      setHearts(heartsCounter + 1)
+      sendHeartToApi()
+    }
+  }
```

Comment:
Great job with heart button. Nice short & clear code. I really like that it's only possible to click once.